front-end-tools/src/constants.ts

```diff
@@ -21,16 +21,33 @@ export const EXAMPLE_JSON_OBJECT = {
 // or does not want to use the embedded schema (meaning if the checkbox "Use module from step 1" is unchecked).
 export const EXAMPLE_ARRAYS = 'Examples: \n\n[1,2,3] or \n\n["abc","def"] or \n\n[{"myFieldKey":"myFieldValue"}]';
 
-// The input parameter can have one of these type options.
+// Available options for the input parameter type in step 2.
+export const INPUT_PARAMETER_TYPE_NUMBER = { label: 'number', value: 'number' };
+export const INPUT_PARAMETER_TYPE_STRING = { label: 'string', value: 'string' };
+export const INPUT_PARAMETER_TYPE_OBJECT = { label: 'object', value: 'object' };
+export const INPUT_PARAMETER_TYPE_ARRAY = { label: 'array', value: 'array' };
+
+// All available options for the input parameter type in step 2.
 export const INPUT_PARAMETER_TYPES_OPTIONS = [
-    { label: 'number', value: 'number' },
-    { label: 'string', value: 'string' },
-    { label: 'object', value: 'object' },
-    { label: 'array', value: 'array' },
+    INPUT_PARAMETER_TYPE_NUMBER,
+    INPUT_PARAMETER_TYPE_STRING,
+    INPUT_PARAMETER_TYPE_OBJECT,
+    INPUT_PARAMETER_TYPE_ARRAY,
 ];
 
 // The subindex of all smart contracts.
 export const CONTRACT_SUB_INDEX = 0n;
 
-// Regular expression of a valid module reference which has to be a hex string `[0-9A-Fa-f]` of length 64.
-export const REG_MODULE_REF = /^[0-9A-Fa-f]{64}$/;
+// Characters of a hex string.
+export const hexRegex = /^[0-9a-fA-F]+$/;
```

Comment:
I wanted to get rid of these hex expressions and rather use the following input `validation`. The input fields are immediately evaluated when the user fills them and an error appears at the front end in case of `wrong` input.

```
    const validateModuleReference = (value: string ) => {
        try {
        
            ModuleReference.fromHexString(value);

            return true;
        } catch (e) {
            return `Invalid module reference. Original error: ${(e as Error).message}`;
        }
    };
    
   ... 
   
   <Form.Control
        defaultValue={MODULE_REFERENCE_PLACEHOLDER}
        ...
        {...form.register('moduleReferenceString', {
               required: true,
              validate: validateModuleReference,
         })}
   />
```

But I observed that the method `ModuleReference.fromHexString(value)` only returns an error if the length of `value` does not match but not if the string characters are not hex. There is no evaluation if the string is actually a hex string since the type `HexString` in our SDK is just an alias for `String`.

```* @throws If the provided hex encoding does not correspond to a buffer of exactly 32 bytes.```
https://github.com/Concordium/concordium-node-sdk-js/blob/main/packages/sdk/src/types/ModuleReference.ts#L92
https://github.com/Concordium/concordium-node-sdk-js/blob/main/packages/sdk/src/types.ts#L35

Suggestion:
- Move back to regular expressions at this front end for now (checking length as well as hexCharacters). 
```
// Regular expression of a valid module reference which has to be a hex string `[0-9A-Fa-f]` of length 64.
export const REG_MODULE_REF = /^[0-9A-Fa-f]{64}$/;
```
- Add an evaluation to our `web-sdk` to check for if the type `HexString` consists of actual hex characters.